### ISSUE TITLE
[v1][QExpansionItem] value prop is ignored when input event is fired 

### ISSUE THREAD

**Issue by behnoodk** [quasarframework/quasar#3735]
### Software version

Quasar: 1.0.0-beta.13
@quasar/app (v1+ only): 1.0.0-beta.13

### JsFiddle
https://codepen.io/anon/pen/NJQOMO

### What did you get as the error?
When I use `:value` with `@input` to control whether the content should hide or not when `input` event is fired, `QExpansionItem` hides/shows the content regardless if I change the model in the `input` handler or not.

### What were you expecting?
If I do not change the model, to `false` for example, `QExpansionItem` should not hide the content.
The behavior is normal when the model is changed.

### What steps did you take, to get the error?
Code sample included in the pen.

**Comment by mountash** [comment]
PR #3819 created

**Comment by smolinari** [comment]
This is interesting and great you put in the effort to try to fix it. 

This is my take. The separation of v-model into the value prop and the `@input` event isn't there to be able to control the toggle. That is a feature within the component. In other words, if you control the value, you control the toggle effect. 

In other words, if you break v-model, by using the value prop with the `@input` event, then it is up to you to make sure the synchronization of the value happens. The expanding and contracting is still going to work as designed. 

If you want to control the toggle, just control the model, as you were doing with value or can do with v-model. If you need to control something happening at the time the content toggles, you have a couple of choices. 

https://codepen.io/smolinari/pen/EJgaxe?editors=1010

And seeing that, I notice the collapsing of the item over v-model isn't firing the `@input` event and that is a genuine problem. 

Scott

**Comment by mountash** [comment]
Hi @smolinari,

I thought that the issue here was that, if you extra split the model into `:value` and `@input` and do not update the model in handler, the list must not collapse if it was opened. 
@behnoodk correct me if I'm wrong.
I don't know for what purposes it is needed, but for example if a user don't have rights to collapse the item 🌝 

If you do the same for the `q-toggle`:
` <q-toggle :model="modelExpanded" @input="handleModelExpandedToggle" label="Expanded"/>`
it won't update the `modelExpanded` if you don't do that in `handleModelExpandedToggle`.
And while `v-model=”dataProperty”` is just short-hand for `:value="dataProperty" @input="dataProperty = $event.target.value"` if you don't extra redefine the `@input`, you will have a model update there.

**Comment by smolinari** [comment]
>if you extra split the model into :value and @input and do not update the model in handler, the list must not collapse if it was opened

The "feature" of toggling the item content is built into the component and is controlled externally by the `value` prop. true=expanded. false=collapsed. The handler is also only a method to catch when the item is expanding or collapsing and if needed, update state. It's not a control to expand or collapse the item. 

> If you do the same for the q-toggle:
<q-toggle :model="modelExpanded" @input="handleModelExpandedToggle" label="Expanded"/>
it won't update the modelExpanded if you don't do that in handleModelExpandedToggle.

There is a difference, but not what you are saying. In QToggle, the toggling feature doesn't work without v-model being set. In fact, for QToggle the `value` prop is required. For QExpansionItem, it isn't. 

Notice no value or v-model here. 

https://codepen.io/smolinari/pen/bJwEar?&editable=true&editors=101  

In other words, in QExpansionItem, v-model is there to toggle the item externally, but isn't necessary. In QToggle, v-model is there to offer the state of the toggle and toggle it and is required. It's two different use cases. 
 
Scott

**Comment by mountash** [comment]
Yep, I understand that these are two different components and they even extending different mixins with different behaviour. 😄 
But the issue descriptions says:
> When I use :value with @input to control whether the content should hide or not when input event is fired, QExpansionItem hides/shows the content regardless if I change the model in the input handler or not.

So, as I understand, user wants to prevent expanding or collapsing by overriding the `@input`. 
And I do not know why such a desire arises 😃

**Comment by behnoodk** [comment]
>  I do not know why such a desire arises

I have a few complex forms with so many sections and sub sections which cannot be divided into different pages so I'm trying to tidy up the form by hiding the parts user usually don't need to fill using `QExpansionItem`. However, when user enters in one of the inputs inside a `QExpansionItem`, I do not want them to be able to hide that content anymore because the forms have complex validations and one depends on another and ... you get the idea.

The reason I filed this issue is that, I expected, as with all other Quasar components and as is standard, when I do not use the `v-model` directive and use the `:value` `@input` pattern, the component should respect the `@input` event and don't hide/open if I don't change the value inside event handler.

As it is now, it hides even though the model is still true, and that is the problem.

**Comment by mountash** [comment]
So my proposal should work for this case. It won't break the common cases functionality, but will give an opportunity to implement such complex forms.

**Comment by smolinari** [comment]
If that is the case, you could use the `disable` prop. 

Scott

**Comment by behnoodk** [comment]
@smolinari Thanks! I should have thought of that myself 😄 
I just tried it and it takes care of disabling the hide/show functionality. Adds some unwanted styling which I can take care of with my own css overrides.

**Comment by smolinari** [comment]
Yeah. I understand what you mean. 

It's a design decision in the end and only @rstoenescu knows what the design was meant to do. 

Looking at QMenu, it also works differently with v-model. v-model allows the menu to expand, but doesn't collapse it or rather something makes it expand again. Hmm....

https://codepen.io/smolinari/pen/XQjKwE?editors=1010

Scott

**Comment by behnoodk** [comment]
> It's a design decision in the end and only rstoenescu knows what the design was meant to do.

Right. Probably have to wait until Razvan finds some time to look at this and tell us hist verdict.

**Comment by rstoenescu** [comment]
The model is not ignored, but there is an inner state too. This is used for cases where there is no v-model attached to the component. This gets toggled no matter what you do in the `@input` handler.

The way to go is to create the `@input` handler like this:

```js
handleToggle (newState) {
    // we set the new state
    this.expanded = newState

    // on next tick (very important), we reset it
    this.$nextTick(() => {
      this.expanded = !newState
    })
  }
```

**Comment by rstoenescu** [comment]
https://codepen.io/rstoenescu/pen/VOBrgd
